Source/Modules/python.cxx

```diff
@@ -396,6 +398,11 @@ class PYTHON:public Language {
 	  py3 = 1;
 	  Preprocessor_define("SWIGPYTHON_PY3", 0);
 	  Swig_mark_arg(i);
+	} else if (strcmp(argv[i], "-py3-stable-abi") == 0) {
+	  py3_stable_abi = 1;
+	  py3 = 1;
```

Comment:
Should this actually imply `-py3`?

You normally get Python3-compatible wrappers without `-py3` and my understanding was `-py3` mostly turned on things of debatable usefulness (such as annotations which aren't very useful as they use the C/C++ type).  See #1779 where William discusses this a bit.

Unless `-py3` is actually required it seems the user ought to get to choose; if it is required, it'd be good to note something like:

      -py3-stable-abi - Generate code compatible with Python 3 stable ABI (PEP 384) (implies -py3)

Reply:
The code using stable API won't compile with Python 2 anyhow, so it doesn't seem to make much sense to use it without `-py3`.

Reply:
My point is that `-py3` (at least currently) doesn't just say "generate wrappers which only work with Python3" but enables things like those annotations which aren't wanted by most people.  There's then no way to turn them off again.

`-py3` really needs sorting out, but chaining more options to it if anything makes that a bit harder to do.

Reply:
Sure, we could remove it from here, it doesn't matter to me personally as I always use both options together anyhow. It just looks like `-py3-stable-abi` should be a superset of `-py3` but, again, I don't care enough about this to argue.